src/compiler/types.ts

```diff
@@ -345,17 +349,31 @@ module ts {
         // error.
         ThisNodeHasError = 1 << 5,

```

Comment:
Not sure but I'd prefer if we did left-shifts because it's less easy to make a mistake in